### ISSUE TITLE
Possibility to send the list of workflow files to lint ?

### ISSUE THREAD

**Issue by nvuillam** [rhysd/actionlint#310]
Hi,

First, thanks for your great linter ! :)

On MegaLinter, we have an issue about filtering -> https://github.com/oxsecurity/megalinter/issues/2295

It seems that even when we send a list of files to actionlint, all the workflows are analyzed anyway

Is the linter able to accept a list of files ?

If not , would it be possible to add it as a new feature ?

Many thanks & best regards !

**Comment by rhysd** [comment]
> Is the linter able to accept a list of files ?

Yes, it should be possible. If file paths are passed to `actionlint` command, the command should check only the files. Did you pass directories to the command? `actionlint` checks all files in a directory if some directory path is given.

Can you show some example of your command line arguments which are passed to `actionlint` command which reproduce this issue?

**Comment by nvuillam** [comment]
@msha1026 @chester-leung please can you run MegaLinter with `LOG_LEVEL: DEBUG` and share the log, or the part of the log when actionlint is called ?

Thanks :)

**Comment by bdovaz** [comment]
> Yes, it should be possible. If file paths are passed to `actionlint` command, the command should check only the files. Did you pass directories to the command? `actionlint` checks all files in a directory if some directory path is given.
> 
> \# Check 2 files only
> actionlint /path/to/workflow1.yaml path/to/workflow2.yml
> 
> \# Check all files in the directory (dir1/*, dir2/*)
> actionlint /path/to/dir1 path/to/dir2
> Can you show some example of your command line arguments passed to `actionlint` command which reproduced this issue?

@rhysd Is it possible to scan a directory? There is no example of this, only of files:

https://github.com/rhysd/actionlint/blob/v1.7.11/docs/usage.md#actionlint-command

Similarly, when you do not provide any arguments, what it's documented is not correct:

> With no argument, actionlint finds all workflow files in the current repository and checks them.

What it actually does is search for the repository and scan the `.github/workflows` path from the root:

https://github.com/rhysd/actionlint/blob/3c8e03070492263721fccf223f691ef34bb0e79c/command.go#L103

https://github.com/rhysd/actionlint/blob/3c8e03070492263721fccf223f691ef34bb0e79c/linter.go#L264
